connectors/es/management_client.py

```diff
@@ -225,15 +224,43 @@ async def index_exists(self, index_name):
             partial(self.client.indices.exists, index=index_name)
         )
 
-    async def get_index(self, index_name, ignore_unavailable=False):
-        return await self._retrier.execute_with_retry(
+    async def get_index_or_alias(self, index_name, ignore_unavailable=False):
+        """
+        Get index definition (mappings and settings) by its name or its alias.
+        """
+        # Example structure of response:
+        # {
+        #     "my-data-index": {
+        #         "aliases": {"search-my-data-index": {}},
+        #         "mappings": { ... },
+        #         "settings": { ... },
+        #     }
+        # }
+        get_index_response = await self._retrier.execute_with_retry(
             partial(
                 self.client.indices.get,
                 index=index_name,
                 ignore_unavailable=ignore_unavailable,
             )
         )
 
+        if index_name in get_index_response:
+            logger.debug(f"Got index by its name: {index_name}")
+            return get_index_response[index_name]
+
+        for (
+            existing_index_name,
+            existing_index_definition,
+        ) in get_index_response.items():
+            if "aliases" not in existing_index_definition:
+                continue
+
+            if index_name not in existing_index_definition["aliases"]:
+                continue
```

Comment:
Nit: it feels not ideal that we explicitly `continue` if not index name not found in aliases, instead we could check if index_name is in aliases and immediately return the definition? 

That would be clearer when and what is returned IMO but nothing blocking

Reply:
Sorry I don't understand the difference - it's same approach but slightly moving pieces around without any real difference?

Reply:
Yeah, it’s mainly about readability, but that’s pretty subjective. Nothing blocking

Reply:
It's a simple change, so I just went ahead and did it :)